sips/reduce_signatures_no_fork.md

```diff
@@ -0,0 +1,129 @@
+|     Author     |            Title            | Category |       Status        |    Date    |
+| -------------- | --------------------------- | -------- | ------------------- | ---------- |
+| Matheus Franco | Reduce Signatures - No Fork | Core     | open-for-discussion | 2024-03-19 |
+
+## Summary
+
+This SIP aims to reduce cryptography costs with a no-fork approach. It accomplishes that by proposing to do only one BLS verification for the quorum of partial signature messages and by stopping relying on BLS for consensus messages.
+
+## Motivation
+
+Cryptography is, commonly, the biggest processing overhead in distributed systems. In our case, the BLS scheme is used which is also one of the heaviest cryptography protocols. Thus, performing one BLS verification for each in-committee message represents an important scalability barrier.
+
+## Rationale
+
+### 1st change
+
+A partial signature message implies one RSA verification (in the message validation layer) and two BLS verifications (for `SignedPartialSignatureMessage` and `PartialSignatureMessage`). Notice that two signatures are used to verify that the sender is correct and one is used to verify if the (same) sender correctly signed the beacon partial signature. Therefore, we can, simply, ignore the first BLS signature since the RSA signature is already being checked.
+
+> [!NOTE]
+> Since we want a no-fork change, one must still BLS-sign twice to be compatible with the previous version.
```

Comment:
There is still a rogue key attack vector here ;-\

Reply:
How is that? Can you give an example?

Reply:
Can't an attacker put a bad sig on the first BLS check simply?

Reply:
Yes, it can. But let's say one malicious operator creates an invalid signature on the `SignedPartialSignatureMessage`. What is the attack outcome in generating an inconsistent state?